src/Storages/HDFS/StorageHDFS.cpp

```diff
@@ -202,7 +202,7 @@ class HDFSSink : public SinkToStorage
         const CompressionMethod compression_method)
         : SinkToStorage(sample_block)
     {
-        write_buf = wrapWriteBufferWithCompressionMethod(std::make_unique<WriteBufferFromHDFS>(uri, context->getGlobalContext()->getConfigRef()), compression_method, 3);
+        write_buf = wrapWriteBufferWithCompressionMethod(std::make_unique<WriteBufferFromHDFS>(uri, context->getGlobalContext()->getConfigRef(), 0), compression_method, 3);
```

Comment:
Does it make sence to support for `StorageHDFS` too? You can just add a setting to `Core/Settings.h` and access it like `context->getSettingsRef().hdfs_dfs_replication`.